api/cluster/controller.go

```diff
@@ -280,6 +305,12 @@ func (c *controller) Delete(ctx context.Context, modelService *models.Service) (
 		}
 	}
 
+	vsName := fmt.Sprintf("%s-%s-%s", modelService.ModelName, modelService.ModelVersion, models.VirtualServiceComponentType)
```

Comment:
I believe we should only attempt to delete this for the new model services with RevisionID > 1. For the older ones, deleting the isvc will handle this VS deletion and this line will return an error, right?

Reply:
Updated as suggested. Thanks, Krithika!